client/docker.go

```diff
@@ -0,0 +1,104 @@
+package client
+
+import (
+	"context"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/mount"
+	"github.com/docker/docker/client"
+	"github.com/docker/docker/pkg/stdcopy"
+	"github.com/jjauzion/ws-worker/conf"
+	"github.com/jjauzion/ws-worker/internal/logger"
+	"go.uber.org/zap"
+	"io"
+	"os"
+)
+
+type DockerHandler struct {
+	client client.APIClient
+	log    *logger.Logger
+	config conf.Configuration
+}
+
+func (dh *DockerHandler) new(log *logger.Logger, config conf.Configuration) error {
+	dh.log = log
+	dh.config = config
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		dh.log.Error("", zap.Error(err))
+		return err
+	}
+	dh.client = cli
+	return nil
+}
+
+func (dh *DockerHandler) runImage(ctx context.Context, image string) error {
+	dh.log.Info("running container", zap.String("image", image))
+	reader, err := dh.client.ImagePull(ctx, image, types.ImagePullOptions{})
+	if err != nil {
+		dh.log.Error("failed to pull image", zap.Error(err))
+		return err
+	}
+	dh.log.Info("docker says:")
+	io.Copy(os.Stdout, reader)
+	//buf := new(strings.Builder)
+	//str := strings.Split(buf.String(), "\n")
+	//for _, s := range str {
+	//	dh.log.Info("docker says", zap.String("", s))
+	//}
```

Comment:
```suggestion
	buf := new(strings.Builder)
	io.Copy(buf, reader)
	str := strings.Split(buf.String(), "\n")
	for _, s := range str {
		if s != "" {
			simple := struct {
				Status string `json:"status"`
			}{}
			err := json.Unmarshal([]byte(s), &simple)
			if err != nil {
				return fmt.Errorf("cannot unmarshal: %w", err)
			}
			dh.log.Info("docker pull_" + simple.Status)
		}
	}
```
<img width="593" alt="Screen Shot 2021-03-21 at 9 21 49 PM" src="https://user-images.githubusercontent.com/19614916/111919673-7e5c5680-8a8b-11eb-85e3-ba32cdd90721.png">